canonical-web-and-design/maas-ui
# refactor(ui): make SSH key components reusable

## Done

- Make the SSH key list and form reusable.
- Migrate the settings table to TypeScript.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit `/MAAS/r/account/prefs/ssh-keys`.
- Check that the table appears.
- Click the import button and the form should appear.

## Fixes

Fixes: canonical-web-and-design/app-squad#163.
